### PR TITLE
Horrorforms for the nth time: Betrayal Edition!

### PR DESCRIPTION
**Pull request by ghost**
:cl:
add: Changelings will ascend to greater forms if they absorb all of their fellow changelings
/:cl:

I love changelings and I've been on this mighty ass quest to do the right thing and really bring changeling back to its source. Those attempts have failed because to make the changeling true to its source you have to throw away all ideas of balance because the thing is absolutely OP. This is probably maybe my last attempt, locking it behind betraying your team and managing to kill all of them to reach the form. Now this does give me more freedom to get the thingling I want, so expect some server ending hooligans in this pr.

I have some uncommitted changes, this pr is ready for testmerge tonight. I also need to add changeling objective to absorb all other changelings and finally, objectives for lesserlings. Cocoon, tendril grabs, assimilate and all that is done.

I NEED SOUNDS FROM THE MOVIE THE THING IF YOU CAN HELP RECREATE THOSE PM ME ON DISCORD.

### CODE REVIEW

File: code/modules/antagonists/changeling/changeling.dm [hunk 1]
```diff
@@ -54,7 +55,7 @@
 		honorific = "Ms."
 	else
 		honorific = "Mr."
-	if(GLOB.possible_changeling_IDs.len)
+	if(GLOB.possible_changeling_IDs.len) //remind me to add lesserling check, we don't want converts to immediately steal all the names creating a division of real names and numerals
```
Comment (Armhulen) [inline code/modules/antagonists/changeling/changeling.dm]: Ignore this I got it done.

**Comment by Aloraydrel** [comment]
BOYS ARE BACK IN TOWN!

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+		M.speed = initial(M.speed)
+
+/obj/item/restraints/handcuffs/horrorform
+	name = "writhing tentacles"
+	desc = "appendages from a hideous creature."
+	icon = 'icons/mob/changeling.dmi'
+	icon_state = "cordgrab"
+	item_flags = DROPDEL
+
+/obj/item/restraints/handcuffs/clockwork/dropped(mob/user)
```
[the remark this answers is not in the document]
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: This was fixed too

**Comment by Armhulen** [comment]
Yeah recreate not steal

**Comment by Therapist1** [comment]
Lookin fresh, man.

**Comment by Armhulen** [comment]
As I kind of touched on in my op, i wouldn't review the code right now.

**Comment by Tlaltecuhtli** [comment]
:tentacle:

**Comment by improvedname** [comment]
>sprite icons taken from thaumcraft minecraft mod.
you have been busted armhulen

**Comment by Armhulen** [comment]
Wait which one holy shit

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+/mob/living/simple_animal/hostile/true_changeling/death()
+	..(1)
+	new /obj/effect/gibspawner/human(get_turf(src))
+	if(stored_changeling && mind)
+		visible_message("<span class='warning'>[src] lets out a furious scream as it shrinks into its human form.</span>", \
+						"<span class='userdanger'>We lack the power to maintain this form! We helplessly turn back into a human...</span>")
+		stored_changeling.loc = get_turf(src)
+		mind.transfer_to(stored_changeling)
+		stored_changeling.Unconscious(300) //Make them helpless for some time
+		stored_changeling.status_flags &= ~GODMODE
```
Comment (octareenroon91) [inline code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: Move this up a line or the unconscious ling would not be exactly 'helpless'.

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: It's not a sleep don't worry BUTT

I should really gut the code anyways, once the ling goes horror it should not go back to human

**Comment by chesse20** [comment]
add this ppplzzz changelings are so underpowered and boring to play as

**Comment by Ratvar** [comment]
As cool as it sounds, what stops a changeling from absorbing clone of a ling to make 5 roundenging antags every time you see them things? What traitors should do in traitorchan,  can't really compete?

Maybe drop whole "ton of hiveminded traitors with nodrop eswords have to steal RD's armor" because it's goddamn lame, and make changeling into alone roundending antag, wizard-blob style?

**Comment by Ratvar** [comment]
https://docs.google.com/document/d/1adKwrrCT_4sHfFDtKfX4Umdf-1cm6B9uZrFn7fJDq9M/edit#
thanks xhuis

**Comment by Armhulen** [comment]
>absorbing clone of a ling

explain?

**Comment by PKPenguin321** [comment]
no you stupid nobody wanted this in the thread you made

**Comment by Armhulen** [comment]
i have four thumby uppies and eight honking horns, i think your :-1: stands no chance. sit yourself.

**Comment by Armhulen** [comment]
also rip the being :*(

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 3 above)
Comment (octareenroon91) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: Right, silly me.

**Comment by Ratvar** [comment]
Changeling №1 walks into cloner.
Changeling №2 sucks changeling №1.
Changeling №2 is now conversion antag thing.
Changeling №2 clones changeling №1.
Changeling №1 sucks someone else blahblah is now conversion antag thing.
+Xenoling going full mindslave++ on everyone.

**Comment by Shadowflame909** [comment]
Wouldn't someone be banned for that... Blatant feature abuse. As easy as an admin going. Bwoink, why are all ling's alive. How did you get horror form? In my opinion the easier to root out cheaters and powergamers. The better. Leave it in armhulenn. Maybe add a roundstart message like drones used to have At most.

**Comment by Dorsisdwarf** [comment]
We dont ban for creative use of game mechanics you dorkus, only actual bug exploitation

**Comment by Armhulen** [comment]
How about I just not make that happen? You can't assume a feature is bad because the creator didn't think of an exploit for it, just fix the exploit

AND I DID THINK OF THAT!

Anyways i'm redoing the action buttons so the testmerge was delayed, sorry peeps

**Comment by Armhulen** [comment]
You become a potentially round ending force of nature, I might consider making mute sting mute hivemind and a ability to get the names of other changelings BUT like I said, this should be an extremely difficult task

**Comment by fluffe9911** [comment]
I think absorbing all other changelings is hard enough without having to rely on RNG but whateves lol

**Comment by Armhulen** [comment]
I'd much rather have this feature be too rare and crazy when it happens then be too common and a round ruining experience

**Comment by fluffe9911** [comment]
Alrighty that seems fair

**Comment by Ratvar** [comment]
Maybe, if one ling absorbes other, every ling can pinpoint ling hunter, so it's harder?

**Comment by Armhulen** [comment]
Okay but then it becomes pda sec funtime

**Comment by Armhulen** [comment]
I'm only gonna go as far as announcing a level something biohazard when it's down to the last changeling 1v1 so the station has some vivid idea they're about to get fucked

**Comment by Iatots** [comment]
I wouldn't see it past some players to just throw their round if it means shutting their hunter out of their goal.

File: code/modules/antagonists/changeling/changeling.dm [hunk 4]
```diff
@@ -73,7 +74,10 @@
 	if(give_objectives)
 		if(team_mode)
 			forge_team_objectives()
-		forge_objectives()
+//		if(lesserling)
```
Comment (ShizCalev) [inline code/modules/antagonists/changeling/changeling.dm]: ded code?

**Comment by Shadowflame909** [comment]
Y'know, I'd still remove the RNG aspect. Considering that it's secs goal to eliminate lings and they usually do it by gibbing and cremating them. This just makes it the goal for lings themselves. Doing all that work, avoiding sec and the chaplain or a random tider only to be left with bad rng would really stink and leave a sour feeling. ALL THE WORK WITH NO PAY-OFF!

**Comment by Armhulen** [comment]
**New in this commit:**

Fixes!
Chaining assimilate from tendril grab is smoother.
Cocoon no longer places a tile ontop of the changeling, fluff improvements
You can no longer try and horrorform multiple times, jesus

Added the lesserling protect objective. Right now there is a runtime, so no testmerges right now!
Finally finished gutting away the old reform code.

**Next up**
tendrils muting the victim so the horrorform doesn't get mobbed immediately in it's creation. BUT THIS TIME I'LL DO IT PROMISE. Fix that damn runtime. besides that i dunno probably just touch up some beam sprites.

**Comment by Ratvar** [comment]
Maybe make tendril grabbing a wall grapple-hook horrorform into that wall?

**Comment by Armhulen** [comment]
Same problem as spine crawl had, thematically nice but that doesn't really serve much purpose for the thingling/it's too niche to be an ability

**Comment by Ratvar** [comment]
Creep that let's changeling walk on it like on magboots and lamely slows normal people? Meaty alien weeds

**Comment by Armhulen** [comment]
I already have this idea, but I'm giving it to lesser changelings so as the station gets taken over the changelings can also spread meatfloor. Seems fun!

As for the rest of the code, I just need to fix one runtime and we're set for testmerge

File: code/modules/antagonists/changeling/changeling.dm [hunk 5]
```diff
@@ -462,7 +463,7 @@
 	hud.leave_hud(owner.current)
 	set_antag_hud(owner.current, null)
 
-/datum/antagonist/changeling/lesser/proc/forge_lesserling_objectives()
+/datum/antagonist/changeling/lesser/proc/forge_lesserling_objectives() //problem: doesn't know what current is. runtimes.
```
Comment (Armhulen) [inline code/modules/antagonists/changeling/changeling.dm]: aay any maintainers wanna help? as the comment says the current runtimes

**Comment by Armhulen** [comment]
**New in this commit:**

Horrorform objective added! You need to absorb three other valid changelings (so, they need to be not xenobio or lesser)

You can no longer buy horrorform but not cast it, it just denys you buying outright.

Cocoon sprites are at least not copied alien sprites, which might have confused people.

Cocoon now has some fun effects while you're hatching.

Tendril grab and assimilate mute the target.

Some changes to tendril grab's beam!

**Comment by Ratvar** [comment]
Horrorform objective? Won't it fail horribly if any other ling gets randomly gibbed/shuttled/burned/monkeysharded/soulsharded/lavalanded? Maybe make it 2 out of 3? So you still have a chance if 1 got caught, and so one left changeling can cooperate with his new boss to make everything on station tasty by backstabbing, but it's still hard. More betrayal.

**Comment by Armhulen** [comment]
Changing objectives have been historically difficult and this is no different. It's a rare objective for a rare outcome

**Comment by davethwave** [comment]
Well on most rounds there are at LEAST 4 lings and usually more. So what you need to be more worried about is if they can fight back enough to get away and alert the other lings.

**Comment by Ratvar** [comment]
oh, somehow magicaly misread it as all lings

**Comment by Armhulen** [comment]
Huge update coming up soon, lots of sounds coming in here.

**Comment by Armhulen** [comment]
**New in this commit:**

True form as a new ability to replace spine crawl: Screech!
High cooldown ability that blinds, mutes and jitters everyone around the true changeling. Right now the true changeling is hilariously bad at dealing with groups so this is my change to that, if you catch it off guard AND it's used it's screech then it deserves to get merked but otherwise some crowd control is always nice

**Comment by Shadowflame909** [comment]
If a ling absorbed a lesser shadowling = Arm-lings? 🤔 

I dig the shadowling similarities!

**Comment by Armhulen** [comment]
some sprites are coming slow so just be patient with me

**Comment by Nabski89** [comment]
I'm doing side sprites for horror1-5, but have gotten side tracked with trying to figure out map loading for my own project lately.

File: code/modules/antagonists/changeling/changeling.dm [hunk 6]
```diff
@@ -468,7 +468,7 @@
 	var/datum/objective/protect/new_objective = new /datum/objective/protect
 	new_objective.owner = owner
 	new_objective.target = master
-	new_objective.explanation_text = "Protect [master.current.real_name], the horrorform."
+	new_objective.explanation_text = "Protect the horrorform." //[master.current.real_name]
```
Comment (chesse20) [inline code/modules/antagonists/changeling/changeling.dm]: why would children/followers of something call it a horrorform. should say protect your mother changeling or somethin

**Comment by Armhulen** [comment]
pretty rude to just let these stagnate when i'm leaving

**Comment by Armhulen** [comment]
also rude to get all these people to help and go but that's a different story

**Comment by Iamgoofball** [comment]
its aight i can cover for you here

**Comment by chesse20** [comment]
unclose and merge this plz

**Pull request closed without merging**